source/common/network/io_socket_handle_impl.cc

```diff
@@ -477,18 +477,23 @@ IoHandlePtr IoSocketHandleImpl::accept(struct sockaddr* addr, socklen_t* addrlen
 Api::SysCallIntResult IoSocketHandleImpl::connect(Address::InstanceConstSharedPtr address) {
   auto sockaddr_to_use = address->sockAddr();
   auto sockaddr_len_to_use = address->sockAddrLen();
-#ifdef __ANDROID_API__
+#if defined(__APPLE__) || defined(__ANDROID_API__)
   sockaddr_in6 sin6;
-  if (sockaddr_to_use->sa_family == AF_INET && forceV6ForAndroid()) {
+  if (sockaddr_to_use->sa_family == AF_INET && forceV6()) {
     const sockaddr_in& sin4 = reinterpret_cast<const sockaddr_in&>(*sockaddr_to_use);
 
     // Android always uses IPv6 dual stack. Convert IPv4 to the IPv6 mapped address when
     // connecting.
     memset(&sin6, 0, sizeof(sin6));
     sin6.sin6_family = AF_INET6;
     sin6.sin6_port = sin4.sin_port;
+#if defined(__ANDROID_API__)
     sin6.sin6_addr.s6_addr32[2] = htonl(0xffff);
     sin6.sin6_addr.s6_addr32[3] = sin4.sin_addr.s_addr;
+#elif defined(__APPLE__)
+    sin6.sin6_addr.__u6_addr.__u6_addr32[2] = htonl(0xffff);
```

Comment:
Linux headers on the left, Darwin on the right.

![image](https://user-images.githubusercontent.com/474794/176228592-581b790c-19f3-40c6-b249-ed7902a7eb8b.png)